mypy/checker.py

```diff
@@ -606,6 +360,8 @@ def visit_func_def(self, defn: FuncDef) -> Type:
             else:
                 # Function definition overrides a variable initialized via assignment.
                 orig_type = defn.original_def.type
+                # XXX This can be None, as happens in
+                # test_testcheck_TypeCheckSuite.testRedefinedFunctionInTryWithElse
```

Comment:
I haven't taken a look at that test yet, but did you intend to fix this in this series, or should we make a ticket to fix it?

If the test crashes, we should at least ensure that it does not crash after this PR.